pkg/frontend/querymiddleware/codec.go

```diff
@@ -402,6 +404,21 @@ func (c prometheusCodec) EncodeRequest(ctx context.Context, r Request) (*http.Re
 	return req.WithContext(ctx), nil
 }
 
+func setRequestHeadersFromOptions(req *http.Request, o Options) {
+	if o.ShardingDisabled {
+		req.Header.Set(totalShardsControlHeader, "0")
+	}
+	if o.TotalShards > 0 {
+		req.Header.Set(totalShardsControlHeader, strconv.Itoa(int(o.TotalShards)))
+	}
+	if o.InstantSplitDisabled {
+		req.Header.Set(instantSplitControlHeader, "0")
+	}
+	if o.InstantSplitInterval > 0 {
+		req.Header.Set(instantSplitControlHeader, time.Duration(o.InstantSplitInterval).String())
+	}
```

Comment:
should we also include `o.CacheDisabled` with `Cache-Control` (I saw the decode counterpart does this)

Reply:
Oh I had missed that one, added in 2aefafceb, thanks!